components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/AuthorizationParams.kt

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.appservices.fxaclient
+
+data class AuthorizationParams(
+    val clientId: String,
+    val scopes: Array<String>,
+    val state: String,
+    val accessType: String = "online",
+    val keyParams: AuthorizationKeyParams? = null
+) {
+    fun intoMessage(): MsgTypes.AuthorizationParams {
+        var b = MsgTypes.AuthorizationParams.newBuilder().setClientId(this.clientId)
+                .setScope(this.scopes.joinToString(" "))
+                .setState(this.state)
+                .setAccessType(this.accessType)
+        if (this.keyParams != null)
+            b = b.setKeyParams(this.keyParams.intoMessage())
+        return b.build()
+    }
+
+    // Auto Generated equals and hashCode
+    // Since AuthorizationParams holds an array of scopes
```

Comment:
nit: I don't understand what this comment is trying to tell me, because it says "Auto generated" but you're adding the implementation by hand.